libp2p/RLPXFrameWriter.h

```diff
@@ -0,0 +1,88 @@
+/*
+ This file is part of cpp-ethereum.
+ 
+ cpp-ethereum is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+ 
+ cpp-ethereum is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+ 
+ You should have received a copy of the GNU General Public License
+ along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/** @file RLPXperimental.h
+ * @author Alex Leverington <nessence@gmail.com>
+ * @date 2015
+ */
+
+
+#pragma once
+
+#include <libdevcore/Guards.h>
+#include "RLPXFrameCoder.h"
+#include "RLPXPacket.h"
+namespace ba = boost::asio;
+namespace bi = boost::asio::ip;
+
+namespace dev
+{
+namespace p2p
+{
+
+/**
+ * @brief Multiplex packets into encrypted RLPX frames.
+ * @todo throw when enqueued packet is invalid
+ * @todo use RLPXFrameInfo
+ */
+class RLPXFrameWriter
+{
+	/**
+	 * @brief Queue and state for Writer
+	 * Properties are used independently;
+	 * only valid packets should be added to q
+	 * @todo implement as class
+	 */
+	struct WriterState
+	{
+		std::deque<RLPXPacket> q;
+		mutable Mutex x;
+		
+		RLPXPacket* writing = nullptr;
+		size_t remaining = 0;
+		bool multiFrame = false;
+		uint16_t sequence;
```

Comment:
Yes; will add a comment. Sequences are meant to roll over so an initial value is somewhat useless (it can be random). If I had a favorite number I would use that.